### PR TITLE
RDM-2504 Spring boot 2 upgrade

### PR DESCRIPTION
**Pull request by tzarsmango**
Applied parameter changes and class modifications to make classes compatible with the new version

https://tools.hmcts.net/jira/browse/RDM-2504

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -20,12 +22,16 @@ plugins {
     id "com.github.kt3k.coveralls" version "2.8.2"
     id "com.github.spotbugs" version "1.6.6"
     id 'io.franzbecker.gradle-lombok' version '2.0'
+    id 'net.ltgt.apt' version '0.21'
```
Comment (adr1ancho) [inline build.gradle]: Do designer need `net.ltgt.apt' version '0.21'`

File: build.gradle [hunk 2]
```diff
@@ -60,11 +66,11 @@ ext {
     commonsLang3Version = '3.8.1'
     hamcrestVersion = 1.3
     hibernateValidatorVersion = '6.0.14.Final'
-    junitJupiterVersion = '5.4.0'
-    junitVintageVersion = '5.4.0'
+    junitJupiterVersion = '5.3.1'
+    junitVintageVersion = '5.2.0'
     postgresqlVersion = '42.2.5'
```
Comment (adr1ancho) [inline build.gradle:66]: I reckon we could use the later version of Jupiter

File: build.gradle [hunk 3]
```diff
@@ -161,13 +173,23 @@ subprojects {
...
+            exclude group: 'org.hibernate', module: 'hibernate-core'
+            exclude group: 'org.hibernate', module: 'hibernate-entitymanager'
+        }
         compile group: 'com.zaxxer', name: 'HikariCP', version: '3.3.1'
         compile group: 'org.jooq', name: 'jool-java-8', version: '0.9.14'
         runtime "org.elasticsearch:elasticsearch:${elasticSearchVersion}"
         compile group: 'javax.inject', name: 'javax.inject', version: '1'
 
         compile group: 'com.microsoft.azure', name: 'azure-storage', version: '8.0.0'
+        compile group: 'org.liquibase', name: 'liquibase-core', version: '3.6.3'
```
Comment (adr1ancho) [inline build.gradle]: Is this a compile dependency?

File: repository/build.gradle [hunk 4]
```diff
@@ -1,25 +1,29 @@
...
 }
 
 dependencies {
     compile("org.springframework.boot:spring-boot-starter-data-jpa") {
         exclude group: 'org.apache.tomcat', module: 'tomcat-jdbc'
     }
-    compile group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: '0.6.0'
-    compile "commons-beanutils:commons-beanutils"
+    compile group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: '1.0.4'
+    compile "commons-beanutils:commons-beanutils:1.9.3"
```
Comment (adr1ancho) [inline repository/build.gradle:19]: I don't think we need to say `1.9.3` here

File: repository/build.gradle [hunk 5]
```diff
@@ -1,25 +1,29 @@
...
     compile "org.apache.commons:commons-lang3"
     compile "io.springfox:springfox-swagger2:${springfoxSwaggerVersion}"
     compile "io.springfox:springfox-swagger-ui:${springfoxSwaggerVersion}"
     testCompile "org.springframework.boot:spring-boot-test"
     compile "javax.validation:validation-api:2.0.1.Final"
     compile "org.hibernate.validator:hibernate-validator:${hibernateValidatorVersion}"
     compile "org.glassfish:javax.el:3.0.1-b11"
-    testCompile "org.liquibase:liquibase-core"
     compile group: 'com.vladmihalcea', name: 'hibernate-types-52', version: '2.4.1'
```
Comment (adr1ancho) [inline repository/build.gradle:28]: I thought there were only a few places that need liquibase.  By declaring compile dependency seemed to be a little too much

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/designer/repository/AbstractDefinitionRepositoryDecorator.java [hunk 6]
```diff
@@ -17,16 +18,12 @@ protected AbstractDefinitionRepositoryDecorator(R repository) {
         this.repository = repository;
     }
 
+    /*FORWARDING METHODS*/
     @Override
     public <S extends T> S save(S s) {
         return repository.save(s);
     }
 
-    @Override
```
Comment (adr1ancho) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/designer/repository/AbstractDefinitionRepositoryDecorator.java]: I could be wrong but this file may fail checkstyle

File: build.gradle (hunk 2 above)
Comment (tzarsmango) [reply build.gradle:66]: It failed on the newer version so I reverted back to this version known to work on definition-store

File: repository/build.gradle (hunk 4 above)
Comment (tzarsmango) [reply repository/build.gradle:19]: throws the following error when we omit the version there
```* What went wrong:
Could not determine the dependencies of task ':bootJar'.
> Could not resolve all task dependencies for configuration ':runtimeClasspath'.
   > Could not find commons-beanutils:commons-beanutils:.
     Required by:
         project : > project :application > project :rest-api > project :domain > project :repository
```

**Review by dwaynebailey — APPROVED** [review]
Superficial review, LGTM

File: repository/build.gradle (hunk 5 above)
Comment (tzarsmango) [reply repository/build.gradle:28]: Agreed, reverting to testCompile and removing compile dependency from the parent project

File: build.gradle (hunk 3 above)
Comment (tzarsmango) [reply build.gradle]: Agreed, reverting to testCompile and removing compile dependency from the parent project

File: build.gradle (hunk 1 above)
Comment (tzarsmango) [reply build.gradle]: agreed, removed that one

**Pull request merged**